### PR TITLE
Integrate CRaSH shell (SSHD) with an issue command.

### PR DESCRIPTION
**Pull request by marekdapps**
- Based on 'mike-crshell' branch.
- Upgrade to Apache SSHD 1.3.0
- [CRaSH fork is here](https://github.com/marekdapps/crash) 
- Before merge: Push fork to the remote Maven repository 
- Testing: starts only with parameters set `java -jar corda.jar --local-shell` or with `--sshd`

crshell

### CODE REVIEW

**Comment by mikehearn** [comment]
Nice! Thanks for the contribution Marek. I've assigned Clinton to continue this work and get it through code review.

From taking a quick look, I can see a few tasks remaining:

- [ ] Changing the login splash is easy to do without forking CRaSH: I managed that locally, we just need to override the login.groovy file.
- [ ] Let's pull the upgraded SSHD plugin into Corda (perhaps as a separate module so other CRaSH users can get an upgraded SSH plugin without the rest), as CRaSH is pluggable, that should not be too hard. Together these things should allow us to use regular CRaSH from upstream, even though it's somewhat dead for now.
- [ ] Rather than having a dedicated issue command, let's integrate the RPC command line parser from the mike-rpc-tool branch as a separate command (maybe we can call it "run"). Being able to issue RPCs from the shell is one of the primary features we want.
- [ ] Let's invert the --local-shell flag to be --no-local-shell and make the Driver.kt integration test tool set it by default (or, move it into the config file). Otherwise many users won't realise the shell is there at all!

File: node/src/main/resources/crash/commands/issue.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import org.crsh.cli.Argument;
+// import net.corda.node.cmdshell.ReflectionKt;
+
+
+@Usage("issue cash")
+public class issue extends BaseCommand {
+
+    @Command
+    public String main(@Argument String partyName, @Argument Integer amount) {
+
```
Comment (Clintonio) [inline node/src/main/resources/crash/commands/issue.java]: Unnecessary whitespace.

File: node/src/main/resources/crash/commands/issue.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import org.crsh.command.BaseCommand;
+import org.crsh.cli.Argument;
+// import net.corda.node.cmdshell.ReflectionKt;
+
+
+@Usage("issue cash")
+public class issue extends BaseCommand {
+
+    @Command
+    public String main(@Argument String partyName, @Argument Integer amount) {
```
Comment (Clintonio) [inline node/src/main/resources/crash/commands/issue.java]: Shouldn't we also be able to specify currency as a parameter? Just dollars is a bit limited.

File: node/src/main/kotlin/net/corda/node/Corda.kt [hunk 3]
```diff
@@ -190,3 +211,66 @@ private fun drawBanner() {
 """\____/     /_/   \__,_/\__,_/""").reset().newline().newline().fgBrightDefault().bold().
 a("--- DEVELOPER SNAPSHOT ------------------------------------------------------------").newline().reset())
 }
+
+fun startShell(dir: Path, cmdLineOptions : CmdLineOptions, node: Node, whenDone: () -> Unit) {
+
```
Comment (Clintonio) [inline node/src/main/kotlin/net/corda/node/Corda.kt]: Whitespace.

**Comment by mikehearn** [comment]
I have been developing this PR further. I'll open up a set of subsequent PRs of smaller size to get it all merged in, so closing this one for now.
